src/components/checkbox/checkbox.js

```diff
@@ -171,12 +183,18 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
       }
 
       function render() {
-        if(ngModelCtrl.$viewValue) {
+        if(ngModelCtrl.$viewValue && !isIndeterminate) {
           element.addClass(CHECKED_CSS);
         } else {
           element.removeClass(CHECKED_CSS);
         }
       }
+
+      function setIndeterminateState(newValue) {
+        isIndeterminate = (newValue === false) ? false : true;
+        element[0].indeterminate = isIndeterminate ? true : undefined;
```

Comment:
It's still not clear why you need to set an `indeterminate` property on the DOM element.